src/randombytes.rs

```diff
@@ -41,6 +46,35 @@ pub fn randombytes_uniform(upper_bound: u32) -> u32 {
     unsafe { ffi::randombytes_uniform(upper_bound) }
 }
 
+new_type! {
+    /// `Seed` bytes for the deterministic random functions
+    secret Seed(SEEDBYTES);
+}
+
+/// The `randombytes_buf_deterministic` function stores size bytes into buf indistinguishable from
+/// random bytes without knowing seed. For a given seed, this function will always output the same
```

Comment:
This needs to be documented better. Add a more generic description of what randombytes_deterministic is, when it should be used, that it should not be used outside of testing and so on. You could probably use a lot of the documentation from libsodium when applicable.

We don't really ever want users to use these functions outside of testing since the whole security of their crypto system will end up depending on the seed.

Reply:
I pushed updated docs with a warning up top. Note that the docs from the earlier push are 1:1 with libsodium.